src/types/assert.d.ts

```diff
@@ -0,0 +1,19 @@
+import type { assert as chaiAssert } from 'chai';
+import type { assert as sinonAssert } from 'sinon';
+
+// This represents the type of the rejects function from
+// src/test-utils/assert-methods.js
+// Defined here as a local type to avoid a dependency with that module
+type Rejects = <T>(
+  promiseResult: Promise<T>,
+  errorMessage: RegExp | string,
+) => Promise<T>;
+
+// During tests bootstrap, we expose sinon's assert and merge it with chai's
+// one. Then, on tests, we perform assertions via the global `assert` object.
+// This informs typescript about the existence of a global `assert` object which
+// has the types of both chai's assert and sinon's assert.
+// Additionally, we also add the `rejects` method to it.
```

Comment:
```suggestion
// During tests bootstrap, we expose chai's assert as a global and merge it with sinon's
// one.
```

I think the first sentence of this comment, with these changes, are enough. Otherwise you're restarting the types. I mentioned chai's assert first since that provides the most "important" / common assert methods (`assert.equal` etc.).